htimeseries/htimeseries.py

```diff
@@ -356,6 +360,25 @@ def read(cls, f, format=TEXT, start_date=None, end_date=None):
         result.data = data
         return result
 
+    def _auto_detect_format(self, f):
+        original_position = f.tell()
+        result = self._auto_detect_format_without_restoring_file_position(f)
```

Comment:
Why f.tell and f.seek is needed?  Storing the last offset takes any rule in later steps?
You are auto-detecting the format should only return the format